lib/matplotlib/finance.py

```diff
@@ -816,6 +816,65 @@ def _candlestick(ax, quotes, width=0.2, colorup='k', colordown='r',
     return lines, patches
 
 
+def _check_input(opens, closes, highs, lows, miss=-1):
+    """Checks that *opens*, *highs*, *lows* and *closes* have the same length.
+    NOTE: this code assumes if any value open, high, low, close is
+    missing (*-1*) they all are missing
+
+    Parameters
+    ----------
+    ax : `Axes`
+        an Axes instance to plot to
+    opens : sequence
+        sequence of opening values
+    highs : sequence
+        sequence of high values
+    lows : sequence
+        sequence of low values
+    closes : sequence
+        sequence of closing values
+    miss : int
+        identifier of the missing data
+
+    Raises
+    ------
+    ValueError
+        if the input sequences don't have the same length
+    """
+
+    def _missing(sequence, miss=-1):
+        """Returns the index in *sequence* of the missing data, identified by
+        *miss*
+
+        Parameters
+        ----------
+        sequence :
+            sequence to evaluate
+        miss :
+            identifier of the missing data
+
+        Returns
+        -------
+        where_miss: numpy.ndarray
+            indices of the missing data
+        """
+        return np.where(np.array(sequence) == miss)[0]
+
+    same_length = (len(opens) == len(highs)) and (len(opens) == len(lows)) and\
```

Comment:
I think that `len(opens) == len(highs) == len(lows) == len(closes)` will also work here (python puts in the implicit `and`s.